lib/statsd_middleware.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+class StatsdMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    start_time = Time.current
+    status, headers, response = @app.call(env)
+    duration = (Time.current - start_time) * 1000.0
+
+    controller = env['action_dispatch.request.path_parameters'][:controller]
+    action = env['action_dispatch.request.path_parameters'][:action]
+    status_key = "api.external.request#status=#{status},controller=#{controller},action=#{action}"
```

Comment:
Hey @aub - I know I suggested `api.external.request` as a prefix in the example I gave, but I think something like `api.rack.request` might be better suited.